### ISSUE TITLE
The experimental results can't be reproduced.

### ISSUE THREAD

**Issue by InspirationLaurie** [lehaifeng/DASNet#32]
appreciate your work!

I notice that the following settings in your paper.
1. you take the DANet with the pre-trained resnet50 for semseg as your basic networks and apply a modified WDMC loss for metric learning.
2. you don't mention that the data augmentation strategy is used in your experiments.

I reproduced your code.

Regularly, we calculate accuracy, precision, recall, f1 for every prediction, and take the average value. However, you calculate the whole TP, TN, FN, FP in the whole test sets. The evaluation process is modified as follows.

pre, rec, acc, f1 are initialized by averageMeter

`
class averageMeter(object):
    """Computes and stores the average and current value"""
    def __init__(self):
        self.reset()

    def reset(self):
        self.val = 0
        self.avg = 0
        self.sum = 0
        self.count = 0

    def update(self, val, n=1):
        self.val = val
        self.sum += val * n
        self.count += n
        self.avg = self.sum / self.count
     ....
    embedding_distance_map = single_layer_similar_heatmap_visual(out_embedding_t0, out_embedding_t1, 'l2')
    
    min_, max_ = np.min(p), np.max(p)
    p = (p-min_)/(max_-min_)
    p[p > 0.5] = 1
    p[p < 0.5] = 0
    pre.update(precision_score(g.reshape(256 * 256), p.reshape(256 * 256)))
    rec.update(recall_score(g.reshape(256 * 256), p.reshape(256 * 256)))
    acc.update(accuracy_score(g.reshape(256 * 256), p.reshape(256 * 256)))
    f1_sc.update(f1_score(g.reshape(256 * 256), p.reshape(256 * 256)))
`

I modify the evaluation metric and remain the rest settings of the experiments except the batch_size is 36 on my computer.

 I got the results.

OA 0.9646501922607422, PRE 0.7079651923928499, REC 0.5782524152913715, F1 0.5973189549732922

which is far away from yours.

How could I reproduce your experiments? : )

**Comment by Vijayraven95** [comment]
Have anyone reproduce the claimed results so far ? I couldn't able to do it. Maxiy dice score I can get is around 0.60.

**Comment by liudong315** [comment]
How can I run through the code？

**Comment by swjtulinxi** [comment]
请问有人解决了吗，阈值应该取多少啊。0.5？

**Comment by silence-tang** [comment]
有需要的兄弟可以参考我复现并改进的[DASNet-V2](https://github.com/silence-tang/DASNet-V2)。修复了一些论文源码中的bug，并且加了decoder结构。把loss替换成cosine sim可以达到0.96+的F1_score。
欢迎star~
